File: Questão5.py
class Animal:
    def som(self):
        raise NotImplementedError("Subclasse deve implementar este método.")

class Cachorro(Animal):
    def som(self):
        return "Au Au"

class Gato(Animal):
    def som(self):
        return "Miau"

def emitir_som(animais):
    for animal in animais:
        print(animal.som())

animais = [Cachorro(), Gato()]
emitir_som(animais)  

File: Questão5.go
package main

import "fmt"


type Animal interface {
    Som() string
}


type Cachorro struct{}


func (Cachorro) Som() string {
    return "Au Au"
}


type Gato struct{}


func (Gato) Som() string {
    return "Miau"
}

func emitirSons(animais []Animal) {
    for _, animal := range animais {
        fmt.Println(animal.Som())
    }
}

func main() {
    animais := []Animal{Cachorro{}, Gato{}}
    emitirSons(animais) 
}
